From b70517b768e8f546427a745c5e834bb0b0e98e0c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?SimaMoto=2CRy=C5=8DTa?= <liangtai.s4@gmail.com>
Date: Sun, 3 Mar 2013 12:42:23 +0900
Subject: [PATCH] Detect openssl without using pkgconfig

* Allow to run on FreeBSD that don't have openssl.pc while it has
  libssl.so
* Detect libcompat.so to use ftime(3)
* Allow to specify PACKAGE_VERSION out of GIT repository
---
 configure.ac | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index 3133b7f..dd508e3 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1,6 +1,6 @@
 # autoconf source script for generating configure
 
-AC_INIT([opusfile], m4_esyscmd([doc/git-version.sh]))
+AC_INIT([opusfile], %%PACKAGE_VERSION%%)
 
 AC_USE_SYSTEM_EXTENSIONS
 AC_SYS_LARGEFILE
@@ -53,8 +53,12 @@ AS_IF([test "x$enable_http" != "xno"], [
    AC_DEFINE([OP_ENABLE_HTTP], [1], [Enable HTTP support])
 ])
 AC_SUBST(openssl)
+AC_CHECK_LIB(ssl,SSL_read,
+	[AC_CHECK_HEADERS(openssl/ssl.h,LIBS="$LIBS -lssl -lcrypto",,)],,
+	-lcrypto -lssl)
 
-PKG_CHECK_MODULES([DEPS], [ogg >= 1.3 opus >= 1.0.1 ${openssl}])
+PKG_CHECK_MODULES([DEPS], [ogg >= 1.3 opus >= 1.0.1])
+AC_CHECK_LIB(compat,ftime)
 
 AC_ARG_ENABLE([fixed-point],
   AS_HELP_STRING([--enable-fixed-point], [Enable fixed-point calculation]),,
-- 
1.8.1.3

